src/complete.cpp

```diff
@@ -1187,6 +1187,10 @@ bool completer_t::try_complete_variable(const wcstring &str) {
 ///
 /// \return 0 if unable to complete, 1 otherwise
 bool completer_t::try_complete_user(const wcstring &str) {
+#ifdef __ANDROID__
+    // The getpwent() call does not exist on Android (and user names are not interesting).
+    return 0;
```

Comment:
While this is strictly speaking correct and functionally equivalent to `return false;` please do the latter for clarity. Having said that, I agree with @floam that since Android is derived from Linux it presumably does utilize UIDs and we should therefore be returning something meaningful on that platform if that is possible. If everything doesn't run as UID zero, and there is no other way to map a UID to name, perhaps this should simply synthesize a user name by doing something like `return sprintf(buf, "user#%d", getuid());`. That is obviously just pseudo-code but you get the idea.